libass/ass_shaper.c

```diff
@@ -480,10 +476,16 @@ static hb_font_t *get_hb_font(ASS_Shaper *shaper, GlyphInfo *info)
     if (!m)
         return NULL;
 
+    hb_font_t *hb_font = hb_font_create_sub_font(font->hb_fonts[info->face_index]);
+    if (!hb_font)
+        return NULL;
+
```

Comment:
i’m not sure about this error check

previously `font->hb_fonts[info->face_index]` was checked for `NULL`, now it is directly passed to hb_font_create_subfont. This is indeed safe and always has been save [since this function was introduced](https://github.com/harfbuzz/harfbuzz/commit/defc45be6d75aba4a67fa7814b91b73bad953fe6#diff-98b5fb69fc328126ab634ef52c824b6146b0fd6505c12f51de7e830485d580daR488), but it will return a dummy font instead like most HarfBuzz API (see also: #566)

Currently the returned dummy object results from a `&Null` cast and the check likely work as intended, but this might change at any point

Reply:
This can no longer be null since 1dfb934affee2a4152619bb671a8c82ec5ccad43; if `ass_create_hb_font` returns null, `add_face` will fail.

Reply:
hmm, `ass_create_hb_font` errors checks likely also don’t work

And to correct my previous message: the `Null` hb’s dummy objects are derived from is not `NULL`/`nullptr` but a custom `Null` object iiuc, so checks will never work
Admittedly this ties in to the preexisting issue #566, but adding new know-to-be-useless checks also doesn't seem right

Reply:
Added a `hb_font_is_immutable` check, mirroring how harfbuzz internally checks for this condition; also added the same in `ass_create_hb_font`.